src/register/mcause.rs

```diff
@@ -1,34 +1,75 @@
+// Copyright (c) 2022 by Rivos Inc.
```

Comment:
This one was simple enough that I didn't bother looking. But seeing this, I don't really see the point of the way they have set it up with a separate `reason` struct for the casting and such. Seems a bit overkill to me. For their more complicated CSRs I'll check what they have first.

Reply:
Yeah, it looks like they introduced the separate `reason` to handle only standard values. Anyways, I asked about tock sources because if we are able to keep the riscv crate implementation close enough to tock, we might be able to get them to use this crate as well. Just more active development and participation from more folks. So, I was wondering if you had found some major issues with the way it is done in tock currently overall (not just for mcause).

Reply:
I see. I kinda touched upon it [here](https://github.com/rivosinc/riscv/pull/1#discussion_r997329969), but I'm _really_ not a fan of the way the have CSRs as structs rather than modules. It does help make the documentation a bit cleaner, but I think it makes the codebase far more confusing to use, and causes mixed syntax for accessing the functions. 

It would definitely be nice to get them to use it too, though, but I don't think switching the structure to something that (at least I think) is worse is a reasonable cost.